tests/callbacks/test_quantization.py

```diff
@@ -43,7 +43,8 @@ def test_quantization(tmpdir, observe: str, fuse: bool, convert: bool):
     """Parity test for quant model."""
     seed_everything(42)
     dm = RegressDataModule()
-    trainer_args = dict(default_root_dir=tmpdir, max_epochs=7, gpus=int(torch.cuda.is_available()))
+    accelerator = "gpu" if torch.cuda.is_available() else "cpu"
+    trainer_args = dict(default_root_dir=tmpdir, max_epochs=7, accelerator=accelerator, devices=1)
```

Comment:
```suggestion
    trainer_args = dict(default_root_dir=tmpdir, max_epochs=7, accelerator="auto", devices="auto")
```

Reply:
devices=auto would do multi-gpu training, which is different from the current test behavior

Reply:
So don't change the code? Or do we want to use "auto"?

Reply:
I think just `accelerator='auto'` would be fine?

Reply:
`accelerator='auto'` is testing different things, current logic will test multi process CPU or single gpu, `accelerator='auto'`  will enable this test on multiple gpus . I think the Strategy will be different(not singleDevice anymore) which means the quant_score will be different as it need call collective to get all the tensors

The goal for this test is not to test the accelerator_connector flags, it's to test quantization, I think we should have flag reproduce test on cpu or single gpu

` ``
accelerator = "gpu" if torch.cuda.is_available() else "cpu"
 if accelerator == "gpu":
    trainer_args = dict(default_root_dir=tmpdir, max_epochs=7, accelerator=accelerator, devices=1)
 if accelerator == "cpu":
   trainer_args = dict(default_root_dir=tmpdir, max_epochs=7, accelerator=accelerator)
```
This should be the exactly reproduce.

Reply:
It's slightly different compared to before, but I don't think it impacts the accuracy of this tests

the reason we can not use 'auto': accelerator='auto' is testing different things, current logic will test multi process CPU or single gpu, accelerator='auto' will enable this test on multiple gpus . I think the Strategy will be different(not singleDevice anymore) which means the quant_score will be different as it need call collective to get all the tensors

Reply:
Ah, so I shouldn't change the code because "auto" will do multi-gpu testing, but this test was made for one gpu. Right?